dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/GitHubUtil.java

```diff
@@ -72,12 +75,18 @@ public GHRepository createFork(GHRepository repo) {
         return null;
     }
 
+    /**
+     * Attempt to delete the GitHub repo if there are no pull requests associated with it
+     * @param repo the repo to delete
+     */
     public void safeDeleteRepo(GHRepository repo) throws IOException {
-        try {
-            repo.delete();
-        } catch (IOException e) {
-            throw new IOException("Please verify that the GitHub token " +
-                    "provided has access to deleting repositories.");
+        if (!repo.queryPullRequests().state(GHIssueState.OPEN).list().iterator().hasNext()) {
+            try {
+                repo.delete();
```

Comment:
Need to retry on failure a few times?

Reply:
In which scenario would you expect that? Are you thinking of a network blip or momentary server unavailability? A potential issue with that would be to potentially put pressure on GitHub rate limits.

One thing that I'm considering to open up as an issue is to avoid forking at all if a repo doesn't need a change. We should be able to detect that off of the primary repo and then just skip forking it at the beginning.

Reply:
Yeh, network blips or errors